.github/actions/deploy_v2/action.yml

```diff
@@ -0,0 +1,185 @@
+
+    - name: Check whether the database exists already
+      if: ${{ inputs.environment == 'review_aks' }}
+      shell: bash
+      run: |
+        select_result=$(bin/konduit.sh ${{ env.app_name }} -- psql -xAc "select count(*) from statistic" && echo "konduit_worked=true" >> $GITHUB_ENV) || true
+        echo "select_result: $select_result"
+        filtered_select_result=$(echo "$select_result" | grep "count")
+        echo "filtered_select_result: $filtered_select_result"
+        echo "stats_count=$filtered_select_result" >> $GITHUB_ENV
+
+    - name: Decide whether to setup and restore the database
+      if: ${{ inputs.environment == 'review_aks' }}
+      shell: bash
+      run: |
+        if [ ${{ env.konduit_worked }} -ne true ] || [ "${{ env.stats_count }}" == 'count|0' ]; then
+          echo "Set variable to download and restore sanitised backup to review app database"
+          echo "setup_database=true" >> $GITHUB_ENV
+        else
+          echo "Database already created and restored"
+          echo "DISABLE_DB_SETUP=1" >> $GITHUB_ENV
+        fi
+
+    - name: Terraform init, plan & apply
+      shell: bash
+      run: make ${{ inputs.environment }} ci deploy
+      env:
+        ARM_ACCESS_KEY: ${{ inputs.arm-access-key }}
+        IMAGE_TAG: ${{ inputs.sha }}
+        TF_VAR_azure_credentials: ${{ inputs.azure-credentials }}
+        CONFIRM_PRODUCTION: yes
+
+    - name: Download Latest Sanitised Backup
+      if: ${{ inputs.environment == 'review_aks' && env.setup_database == 'true' }}
+      shell: bash
+      run: |
+        sanitised_backup_workflow_run_id=$(gh run list -w "Database Backup and Restore" -s completed --json databaseId --jq '.[].databaseId' -L 1)
+        echo "Download artifact for workflow run ID: $sanitised_backup_workflow_run_id"
+        gh run download $sanitised_backup_workflow_run_id
+
+    - name: Restore backup to aks env database
+      if: ${{ inputs.environment == 'review_aks' && env.setup_database == 'true' }}
+      shell: bash
+      run: |
+        bin/konduit.sh -i backup_sanitised/backup_sanitised.sql -t 7200 ${{ env.app_name }} -- psql
+
+    - name: Run Smoke Tests for ${{ inputs.environment }}
+      uses: ./.github/actions/smoke-test_v2/
+      with:
+        environment: ${{ inputs.environment }}
+        pr-number: ${{ inputs.pr-number }}
+        slack-webhook: ${{ inputs.slack-webhook }}
+
+    # - name: Alert on Failure
+    #   if: ${{ failure() && github.ref == 'refs/heads/main' }}
+    #   uses: rtCamp/action-slack-notify@master
+    #   env:
+    #     SLACK_CHANNEL: twd_findpub_tech
+    #     SLACK_COLOR: '#ef5343'
+    #     SLACK_ICON_EMOJI: ':github-logo:'
+    #     SLACK_USERNAME: Publish Teacher Training
+    #     SLACK_TITLE: Deploy to ${{ inputs.environment }} Failed
+    #     SLACK_MESSAGE: ':alert: <!channel> Deployment failure on ${{ inputs.environment }} :sadparrot:'
+    #     SLACK_WEBHOOK: ${{ inputs.slack-webhook }}
```

Comment:
Leftovers?

Reply:
Yea, I'll comment out when all out testing is done